src/notebooks/R Mia Examples/Comparative-Metagenomics.qmd

```diff
@@ -0,0 +1,852 @@
+
+1. Let's define the list of methods we're going to test.
+
+```{r}
+# Define method list
+vegdist_methods <- c("manhattan", "euclidean", "canberra", "bray", "kulczynski", "jaccard",
+                     "gower", "altGower", "morisita", "horn", "mountford", "raup",
+                     "binomial", "chao", "cao")
+
+# Create empty vectors to hold method names and plots
+created <- c()
+plots <- c()
+```
+
+2. Loop through these method types and plot abundance for each, saving each plot into the `plots` vector.
+
+```{r}
+# Loop through these methods and plot abundance
+for (method in vegdist_methods) {
+  # Set the method name
+  method_name <- paste("MDS", method, sep = "_")
+  created <- method_name
+  
+    # Run MDS 
+    mds_tse <- runMDS(filtered_tse,
+                  FUN = vegan::vegdist,
+                  method = method,
+                  assay.type = "css",
+                  name = method_name)
+  
+  # Create plot object of reducedDim slot
+  p <- plotReducedDim(mds_tse, dimred = paste("MDS", method, sep = "_"))
+  
+  # Calculate explained variance
+  e <- attr(reducedDim(mds_tse, method_name), "eig")
+  rel_eig <- e / sum(e[e > 0])
+  
+  # Add explained variance for each axis
+  p <- p + labs(x = paste("MDS_1_", method, " (", round(100 * rel_eig[1], 1), "%)", sep = ""),
+                y = paste("MDS_2_", method, " (", round(100 * rel_eig[2], 1), "%)", sep = ""),
+                title = paste("MDS using distance method", method)) +
+    theme_minimal(base_size = 6) + # Use a minimal theme for clarity
+    theme(
+      plot.title = element_text(hjust = 0.5), # Center the title
+      axis.title = element_text(size = 6), # Smaller axis titles
+      axis.text = element_text(size = 6), # Smaller axis labels
+      legend.position = "bottom", # Position the legend at the bottom
+      legend.title = element_text(size = 6), # Smaller legend title
+      legend.text = element_text(size = 6) # Smaller legend text
+    )
+  
+  # Save plot
+  plots[[method_name]] <- p
+}
+```
+
+3. Combine all plots using patchwork.
+
+```{r}
+# Combine all plots using patchwork
+wrapped_plots <- wrap_plots(plots) +
+  plot_layout(guides = "collect") + 
+  plot_annotation(title = "MDS on Various Distance Metrics",
+                  theme = theme(plot.title = element_text(size = 14, hjust = 0.5)))
+```
+
+Separation of the groups observed in some of the ordination plots could be due to location effect, dispersion effect, or both. To test the location effect (null hypothesis of no difference in composition between groups) we use a permanova implemented in the `vegan` library’s `adonis2` function. We will use the distance metric that best segregates the groups and determine whether the two groups of samples have different centroids. 
+
+1. First we create the metadata data frame.
+
+```{r}
+# Create the metadata data frame
+metadata <- data.frame(colData(filtered_tse))
+rownames(metadata) <- colnames(filtered_tse)
```

Comment:
is this step necessary? if the data is OK from start, the rownames should be there already